### PR TITLE
Separate profile endpoint into 3 endpoints: profile, achievements, su…

### PR DESCRIPTION
**Pull request by yattias**
- Separate profile endpoint into 3 endpoints: profile, achievements, summary_stats
- Cache summary_stats, achievements for an hour to optimize (no purge mechanism, just TTL of one hour)

### CODE REVIEW

File: src/user/related_models/author_model.py [hunk 1]
```diff
@@ -130,6 +138,52 @@ def profile_image_indexing(self):
                 return str(self.profile_image)
         return None
 
+    @property
```
Comment (yattias) [inline src/user/related_models/author_model.py:141]: Moved these from serializer into properties on model

**Review by koutst — APPROVED** [review]
LGTM

File: src/user/views/author_views.py [hunk 2]
```diff
@@ -87,6 +88,47 @@ def update(self, request, *args, **kwargs):
...
         return Response(serializer.data)
 
+    @action(detail=True, methods=["get"], permission_classes=[AllowAny])
+    def summary_stats(self, request, pk=None):
+        author = self.get_object()
+        cache_key = f"author-{author.id}-summary-stats"
+        cache_hit = cache.get(cache_key)
+
+        if cache_hit:
+            return Response(cache_hit)
```
Comment (gzurowski) [inline src/user/views/author_views.py:98]: Nit: Just to align with the code further below in the method.
```suggestion
            return Response(cache_hit, status=200)
```

**Review by gzurowski — COMMENTED** [review]
Existing tests are failing, tests for the newly introduced code are missing.

**Comment by yattias** [comment]
@gzurowski will fix tests / add tests once I finish the frontend portion

**Pull request merged**